dali/operators/reader/loader/loader.h

```diff
@@ -228,18 +228,35 @@ class Loader {
     }
     return SizeImpl();
   }
+
   // Give the size of the data accessed externally from python
-  Index SizePadded() {
+  Index SizePadded(bool consider_padding = true) {
     if (!loading_flag_) {
       PrepareMetadata();
     }
-    if (pad_last_batch_) {
+    if (pad_last_batch_ && consider_padding) {
       return num_samples(num_shards_, SizeImpl()) * num_shards_;
     } else {
       return SizeImpl();
     }
   }
 
+  int GetShardNumber() {
```

Comment:
```suggestion
  int GetNumShards() {
```
GetShardNumber sounds like it's returns the number (index) of the shard, not number of shards.

Reply:
Done